### PR TITLE
Fix various TS types

### PR DESCRIPTION
**Pull request by evgenyboxer**
Fix various types

### CODE REVIEW

File: src/services/balancer/contracts/contracts/vault.ts [hunk 1]
```diff
@@ -96,7 +100,8 @@ export default class Vault {
           100
       );
     } else if (type === 'Stable') {
-      return tokens.map(() => 100 / tokens.length);
+      const tokensList = Object.values(tokens);
+      return tokensList.map(() => 100 / tokensList.length);
```
Comment (evgenyboxer) [inline src/services/balancer/contracts/contracts/vault.ts:104]: @garethfuller is this the intended behaviour here?

File: src/services/pool/calculator/index.ts [hunk 2]
```diff
@@ -81,7 +84,7 @@ export default class Calculator {
       amounts.send.forEach((amount, amountIndex) => {
         const greaterThanBalance =
           Number(amount) >
-          this.allTokens[this.tokenOf(type, amountIndex)].balance;
+          Number(this.allTokens[this.tokenOf(type, amountIndex)].balance);
```
Comment (evgenyboxer) [inline src/services/pool/calculator/index.ts:87]: This coercion in JavaScript would work fine, but TypeScript doesn't like it... so gotta convert it to a number first before doing the math operation.

File: src/types/index.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export interface FormRef {
 
 export interface Token {
   address: string;
-  balance: number;
+  balance: string;
```
Comment (evgenyboxer) [inline src/types/index.ts:11]: Have console logged in - and its a string

File: src/services/balancer/contracts/contracts/vault.ts [hunk 4]
```diff
@@ -96,7 +100,8 @@ export default class Vault {
           100
       );
     } else if (type === 'Stable') {
-      return tokens.map(() => 100 / tokens.length);
```
Comment (evgenyboxer) [inline src/services/balancer/contracts/contracts/vault.ts:99]: Tokens was never a an array, it was actually a map.

**Review by garethfuller — APPROVED** [review]

**Pull request merged**